libcudacxx/include/cuda/std/__memory/assume_aligned.h

```diff
@@ -30,22 +31,24 @@
 _LIBCUDACXX_BEGIN_NAMESPACE_STD
 
 template <size_t _Align, class _Tp>
-_CCCL_NODISCARD _LIBCUDACXX_HIDE_FROM_ABI _CCCL_CONSTEXPR_CXX14 _Tp* assume_aligned(_Tp* __ptr) noexcept
+[[nodiscard]] _LIBCUDACXX_HIDE_FROM_ABI constexpr _Tp* assume_aligned(_Tp* __ptr) noexcept
 {
-  static_assert(_CUDA_VSTD::has_single_bit(_Align), "std::assume_aligned requires the alignment to be a power of 2!");
-#if defined(_CCCL_BUILTIN_IS_CONSTANT_EVALUATED) && defined(_CCCL_BUILTIN_ASSUME_ALIGNED)
-  if (!_CCCL_BUILTIN_IS_CONSTANT_EVALUATED())
+  static_assert(_CUDA_VSTD::has_single_bit(_Align), "std::assume_aligned requires the alignment to be a power of 2");
+  static_assert(_Align >= sizeof(_Tp), "Alignment must be greater than or equal to the size of the input type");
+#if !defined(_CCCL_BUILTIN_IS_CONSTANT_EVALUATED)
+  return __ptr;
+#else
+  if (!is_constant_evaluated())
```

Comment:
We should always qualify 
```suggestion
  if (!_CUDA_VSTD::is_constant_evaluated())
```

Reply:
not really. We are already in the right namespace and ADL doesn't apply here

Reply:
no problem in this case. I will fix it

Reply:
`using namespace std;` wants to have a talk with you